hoomd/filter/set_.py

```diff
@@ -0,0 +1,35 @@
+from hoomd.filter.filter_ import ParticleFilter
+from hoomd import _hoomd
+
+
+class ParticleFilterSetOperations(ParticleFilter):
+    def __init__(self, f, g):
+        if f == g:
+            raise ValueError("Cannot use same filter for {}"
+                             "".format(self.__class__.__name__))
```

Comment:
Why not? It should work, right? It just returns an empty list of tags for differences and the same tags as its inputs for intersection and union.

Reply:
The idea was to prevent a user for shooting themselves in the foot. I can't think of a case where this would be useful, but if you feel strongly about it, it will work.